crates/trie/parallel/src/proof.rs

```diff
@@ -287,14 +291,29 @@ where
                     hash_builder.add_leaf(Nibbles::unpack(hashed_address), &account_rlp);
 
                     // We might be adding leaves that are not necessarily our proof targets.
-                    if targets.contains_key(&hashed_address) {
+                    if storages.contains_key(&hashed_address) {
                         storages.insert(hashed_address, storage_multiproof);
                     }
                 }
             }
         }
         let _ = hash_builder.root();
 
+        // Process the rest of storage proofs that did not have an accosiated account proof
+        // requested.
+        for hashed_address in targets.storages.keys() {
+            let rx = storage_proofs.remove(hashed_address).unwrap();
```

Comment:
this will panic

Reply:
I don't fully understand why, can you expand?

Reply:
didn't see https://github.com/paradigmxyz/reth/blob/9c0b1f3ac49b3e32d5a7bd1e5e087dc9f5cba30a/crates/trie/parallel/src/proof.rs#L258